### PR TITLE
[quant][fx] store dtype, axis as literals in the graph

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54860 [quant][fx] Add pass in convert to fold quant-dequant sequence
* #54859 [quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN
* **#54624 [quant][fx] store dtype, axis as literals in the graph**

Summary:
previously we were creating setattr nodes for dtype and axis.
The FX convention is that primitive types are embedded as literals in args/kwargs.

With this change we won't see getattr nodes in the graph anymore for dtype/axis

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27306898](https://our.internmc.facebook.com/intern/diff/D27306898)

### CODE REVIEW

**Comment by jerryzh168** [comment]
do we need "dtype" to appear in state_dict?

**Review by jerryzh168 — APPROVED** [review]

**Comment by supriyar** [comment]
> do we need "dtype" to appear in state_dict?

I don't think so. I think this change existed even before we added support for scale/zero_point to appear in the state_dict. Since we never call register_buffer for dtype, it was never part of state_dict.

**Comment by jerryzh168** [comment]
> > do we need "dtype" to appear in state_dict?
> 
> I don't think so. I think this change existed even before we added support for scale/zero_point to appear in the state_dict. Since we never call register_buffer for dtype, it was never part of state_dict.

so only scale/zero_point need to be loaded from state_dict, dtype is fixed?

**Comment by supriyar** [comment]
> > > do we need "dtype" to appear in state_dict?
> > 
> > 
> > I don't think so. I think this change existed even before we added support for scale/zero_point to appear in the state_dict. Since we never call register_buffer for dtype, it was never part of state_dict.
> 
> so only scale/zero_point need to be loaded from state_dict, dtype is fixed?

Yes, the dtype information is retrieved from the qconfig, so it doesn't need to be stored in the state_dict. As long as we use the same config at load time, it should match.

**Comment by jerryzh168** [comment]
> > > > do we need "dtype" to appear in state_dict?
> > > 
> > > 
> > > I don't think so. I think this change existed even before we added support for scale/zero_point to appear in the state_dict. Since we never call register_buffer for dtype, it was never part of state_dict.
> > 
> > 
> > so only scale/zero_point need to be loaded from state_dict, dtype is fixed?
> 
> Yes, the dtype information is retrieved from the qconfig, so it doesn't need to be stored in the state_dict. As long as we use the same config at load time, it should match.

I see, sounds good.
